File: 0278. First Bad Version.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        min = 0 
        max = n 
        

        
        while max > min:
            mid = min + int((max-min)/2) 
            if isBadVersion(mid):
                max = mid
            else:
                min = mid+1
            
        

        return max

File: 0278. First Bad Version.cs
/* The isBadVersion API is defined in the parent class VersionControl.
      bool IsBadVersion(int version); */

public class Solution : VersionControl {
    public int FirstBadVersion(int n) {
        int min = 0 ;
        int max = n ;
        int mid;

        // 用二分法進行搜尋，減少loop次數
        while(max > min){
            mid = min + ((max-min)/2); 
            if(IsBadVersion(mid)){
                max = mid;
            } else {
                min = mid+1;
            }
        }

        return max;
    }
}